packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -121,9 +131,17 @@ export const createPluginActions: CreatePluginActions = (
         onUpdateState(state)
       },
       selectAsset: () => {
-        return new Promise((resolve) => {
+        if (assetSelectedCallbackId !== undefined) {
+          // eslint-disable-next-line functional/no-promise-reject
+          return Promise.reject(
+            'Please wait until an asset is selected before making another request.',
+          )
+        }
+        const callbackId = getRandomString(16)
```

Comment:
I thought about that, but decided to have a local `callbackId`, because what if.. `selectAsset()` is called twice? Then the `assetSelectedCallbackId` gets replaced. I don't know if it can really happen, but just wanted to make sure to keep the local version in this scope, so that we can post it to container. What do you think? 🤔

Reply:
ah I see what you mean

Reply:
okay I applied your feedback but unfortunately reverted it.

```ts
1        assetSelectedCallbackId = getRandomString(16)
2        return new Promise((resolve) => {
3          assetSelectedCallbackRef = resolve
4          postToContainer(assetModalChangeMessage(uid, assetSelectedCallbackId))
5        })
 ```

`assetSelectedCallbackId` is `string | undefined`. Although we have assigned `getRandomString(16)`, at the lint 4, TypeScript complains it might be still `undefined` 🙃 
It doesn't make sense, but I didn't want to `assetSelectedCallbackId!`, so brought the local variable back.